src/platform/plugins/shared/discover/public/application/main/components/chart/use_discover_histogram.ts

```diff
@@ -214,102 +202,132 @@ export const useDiscoverHistogram = (
     searchSessionId,
     lastReloadRequestTime,
   } = requestParams;
-  // When in ES|QL mode, update the data view, query, and
-  // columns only when documents are done fetching so the Lens suggestions
-  // don't frequently change, such as when the user modifies the table
-  // columns, which would trigger unnecessary refetches.
-  const esqlFetchComplete$ = useMemo(
-    () => createFetchCompleteObservable(stateContainer),
-    [stateContainer]
-  );
 
-  const [initialEsqlProps] = useState(() =>
-    getUnifiedHistogramPropsForEsql({
-      documentsValue: documents$.getValue(),
-      savedSearch: stateContainer.savedSearchState.getState(),
-    })
-  );
+  const dataView = useCurrentDataView();
 
-  const {
-    dataView: esqlDataView,
-    query: esqlQuery,
-    columns: esqlColumns,
-    table,
-  } = useObservable(esqlFetchComplete$, initialEsqlProps);
+  const histogramCustomization = useDiscoverCustomization('unified_histogram');
 
-  useEffect(() => {
-    if (!isEsqlMode) {
-      setIsSuggestionLoading(false);
-      return;
-    }
+  const query = useAppStateSelector((state) => state.query);
+  const appFilters = useAppStateSelector((state) => state.filters);
+  const { filters: globalFilters } = useCurrentTabSelector((state) => state.globalState);
 
-    const fetchStart = stateContainer.dataState.fetchChart$.subscribe(() => {
-      setIsSuggestionLoading(true);
-    });
-    const fetchComplete = esqlFetchComplete$.subscribe(() => {
-      setIsSuggestionLoading(false);
-    });
+  const filtersMemoized = useMemo(() => {
+    const allFilters = [...(globalFilters ?? []), ...(appFilters ?? [])];
+    return allFilters.length ? allFilters : EMPTY_FILTERS;
+  }, [appFilters, globalFilters]);
 
-    return () => {
-      fetchStart.unsubscribe();
-      fetchComplete.unsubscribe();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const timeRangeMemoized = useMemo(() => timeRange, [timeRange?.from, timeRange?.to]);
```

Comment:
Nit: I wonder if we even need `timeRangeMemoized` anymore now that we're only updating params when `fetch` is called?

Reply:
We can remove it, also because it's a part of `dataRequestParams` now.
Addressed in https://github.com/elastic/kibana/pull/241184/commits/cabd7eb1069b735f27223a76f28790f3e726f3bd